### DISCUSSION TITLE
How to update a page property?

### DISCUSSION THREAD

**Discussion by rwaal** [ramnes/notion-sdk-py#159] [Q&A] [answered]
Hi everyone, 

I have a Notion database with pages. In one of the pages I'd like to update the value of a property called "Test"

``` Python
page = notion.pages.retrieve(page_id)
page['properties']['Test']['rich_text'][0]['plain_text'] = "My updated value"
```
So far so good. If I run these two lines in the debugger, I see the variable being updated in the VS Code debugger. But how do I actually send the update back to the Notion API? There seems to be an `update` method, but I can't find any description on how to use it. 

Could anyone give me some pointers? 
Many thanks in advance

**Comment by ramnes** [comment, marked as answer]
Hey there!

Yes, you want to use the `notion.pages.update(...)` method.

You can find its [Python documentation here](https://ramnes.github.io/notion-sdk-py/reference/api_endpoints/#notion_client.api_endpoints.PagesEndpoint.update), and the [reference API documentation here](https://developers.notion.com/reference/patch-page).

TL;DR, you probably want something like this:

```python
notion.pages.update(
    page_id,
    properties={
        "Test": {
            "rich_text": [{"type": "text", "text": {"content": "My updated value"}}]
        }
    },
)

```

**Reply by rwaal** [reply]
Hi @ramnes, thanks allot for your reply. 

I have been trying to implement your suggestion, but I can't get it to work. When running your suggestion I get the following exception:

```
notion_client.errors.APIResponseError: body failed validation. Fix one:
body.properties.Test.rich_text[0].text should be defined, instead was `undefined`.
body.properties.Test.rich_text[0].mention should be defined, instead was `undefined`.
body.properties.Test.rich_text[0].equation should be defined, instead was `undefined`.
```

Is the property that I'm passing incorrect? Or should the properties `text`, `mention` and `equation` be part of the body of the update? If so, why? Because the documentation states:

> Properties that are not set via the properties parameter will remain unchanged.

Therefore I assumed you only need to pass the property (and value) of the stuff you want change. 
My lack of experience with the Notion API and Python are probably why I can't get it work. But perhaps you could give me another pointer.

Thanks again.

**Reply by rwaal** [reply]
Never mind my previous reply, I think I got it working! I used the following to perform a succesfull update:

``` python
notion.pages.update(
    page_id, properties={"Test":{"rich_text":[{"type":"text","text":{"content":"My updated value"},"plain_text":"My updated value"}]}}
)
```

I must say I don't understand why I had to pass the additional properties as part of the body. But perhaps a more extensive read of the Notion API docs will provide more insights. 

Thanks again for your response. And thanks for creating and maintaining the SDK. It's very much appreciated. ❤️

**Reply by ramnes** [reply]
Yeah, Notion's API is super verbose. FWIW I think you can drop the `plain_text` part actually. I copy-pasted it from your original example but `{"type":"text", "text": {"content": "My updated value"}}` seems like the way to go from what I can read in their docs!
